iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnBuffers.cpp

```diff
@@ -535,84 +536,30 @@ LogicalResult PadOpConversion::apply(
   Value paddingVal =
       paddingConstVal
           ? rewriter.create<ConstantOp>(loc, paddingConstVal).getResult()
-          : adaptor.padding_value();
+          : rewriter.create<LoadOp>(loc, adaptor.padding_value());
 
-  auto operandType = adaptor.operand().getType().cast<ShapedType>();
-  int rank = operandType.getRank();
-
-  SmallVector<Attribute, 2> indexingMaps;
-  indexingMaps.emplace_back(getPadOpInputIndexingMap(op, rank, rewriter));
-  if (!paddingConstVal) {
-    indexingMaps.emplace_back(AffineMapAttr::get(
-        AffineMap::get(rank, /*symbolCount=*/0, rewriter.getContext())));
-  }
-  indexingMaps.emplace_back(AffineMapAttr::get(
-      AffineMap::getMultiDimIdentityMap(rank, rewriter.getContext())));
-
-  SmallVector<Type, 2> resultTypes = {};
-  SmallVector<Value, 2> linalgOpArgs = {adaptor.operand()};
-  if (!paddingConstVal) linalgOpArgs.push_back(adaptor.padding_value());
-  linalgOpArgs.push_back(resultBuffers[0]);
-  auto linalgOp = rewriter.create<linalg::IndexedGenericOp>(
-      loc, resultTypes, linalgOpArgs,
-      rewriter.getI64IntegerAttr(linalgOpArgs.size() - 1),  // args_in
-      rewriter.getI64IntegerAttr(1),                        // args_out
-      rewriter.getArrayAttr(indexingMaps),
-      getParallelAndReductionIterAttrs(rewriter, rank, /*nReduction=*/0),
-      /*doc=*/nullptr, /*library_call=*/nullptr);
-
-  // Add a block to the region.
-  auto *region = &linalgOp.region();
-  auto *block = rewriter.createBlock(region, region->end());
-  SmallVector<Type, 4> bodyArgTypes;
-  bodyArgTypes.append(rank, rewriter.getIndexType());
-  bodyArgTypes.append(linalgOpArgs.size(), operandType.getElementType());
-  block->addArguments(bodyArgTypes);
-  rewriter.setInsertionPointToEnd(block);
-
-  // If the `index` of the result at a particular dimension i, is d_i, check if
-  //
-  // (d_i >= edge_padding_low[i]) &&
-  // (d_i < (edge_padding_low[i] + operand_shape[i])).
-  //
-  // If true, then use the value of the operand, otherwise use the padding
-  // value.
   const auto &edgePaddingLow = op.edge_padding_low();
-  const auto &edgePaddingHigh = op.edge_padding_high();
+  const auto &interiorPadding = op.interior_padding();
```

Comment:
So we return error when interior padding contains non-zero values at the very beginning. Here we are actually able to handle that case.. Should we drop the check at the beginning? (Can happen as a separate patch and will want to have tests.)

Reply:
Ah, forgot the condition. I think we can remove it, it was there because it made the logic complicated. It's simple in this approach, let me also add a test for it.

note that it's still passing on Android.